src/reducers/index.js

```diff
@@ -90,15 +101,30 @@ function getInitialState() {
     scrollbarYWidth: Scrollbar.SIZE,
     scrolling: false,
 
-    /*
-     * Internal state only used by this file
-     * NOTE (jordan) internal state is altered in place
-     * so don't trust it for redux history or immutability checks
-     * TODO (jordan) investigate if we want to move this to local or scoped state
+    /**
+     * Internal state is only used by reducers.
+     * NOTE (jordan, pradeep): Internal state is altered in place, so don't trust it for redux history or immutabability checks.
```

Comment:
@wcjordan, pinging you for sanity checking since you have the most context regarding "internal state".

Reply:
Oh man, it's been too long for me to have much of an opinion here.  But no strong concerns from my end...

The only thing I would double check is, my understanding of redux is that it's designed so reducers are pure functions which take state as input and produce the next state as output.  With this change, that's no longer true.  They take the state + the internal state object as input instead.  You may want to think through any assumptions of redux which might be violated by that.

Reply:
@wcjordan , good point. I've been debating myself about this the last couple of days...

Storing internal state outside the redux store almost makes it seem like it's an external factor to our reducers, which _can_ make the reducers seem impure in a sense.
But I'll also argue that _currently_ internal state values are strictly and purely calculated out of input state, and are **not modifiable** anywhere else. 
So internal state could just be seen as functions over input state.
Technically, this makes our reducers remain pure and deterministic.

----

The only other issue that I could think of is how internal state is **mutated**...
This breaks time travel debugging and serialization of state, both of which we haven't been caring about since v1, but I was wondering if anything else breaks?

In our reducers, we assign parts of internal state back to output state (eg: slices of storedHeights are copied over to visible row heights).
From React's/Redux's perspective, output state could be seen as mutated, potentially causing strict equality checks to pass even when data changed.

But so far, this has never been an issue just because the entire subset of data that gets assigned over from internal state to output state is _primitive_ data.

I think we're safe here for now, but we'll need to revisit this once more if our use cases start getting complicated.

Reply:
Got it.  Thanks for your analysis.  I appreciate the detail and sounds like we're on a reasonable track with this change.